src/esrs/policy.rs

```diff
@@ -0,0 +1,41 @@
+use async_trait::async_trait;
+
+use crate::{AggregateManager, StoreEvent};
+
+#[async_trait]
+pub trait Policy<Manager>
+where
+    Self: PolicyClone<Manager>,
+    Manager: AggregateManager,
+{
+    /// This function intercepts the event and, matching on the type of such event
+    /// produces the appropriate side effects.
+    /// The result is meant to catch generic errors.
+    async fn handle_event(&self, event: &StoreEvent<Manager::Event>) -> Result<(), Manager::Error>;
+}
+
+pub trait PolicyClone<Manager>
```

Comment:
I can't really understand what happens in the rest of this file?

Reply:
This is an hack in order to implement the clone trait for every type that is a `Box<dyn Policy>`